### ISSUE TITLE
disable SWA make performance worse

### ISSUE THREAD

**Issue by valencebond** [voyazici/orderless-rnn-classification#3]
thanks for your code. This code uses SWA which is not mentioned in your paper, and i find there is a big performance drop without SWA,  and the performance is same as baseline, i.e. training with bce loss. Is that right? 
please correct me if there is something wrong.

**Comment by voyazici** [comment]
Hi,
In the subsection of network training it is stated that we use stochastic weight averaging (SWA). It is true that the SWA improves the performance, that's the reason we included it. However, to come up with the best combination of hyperparameters for the model with SWA, we launched many experiments on the validation set with different hyperparameters. Since it is a combined architecture (encoder-decoder) with different training steps, the final performance is sensitive to the hyperparameters. You should find the optimal hyperparameter values for the model without SWA in the same manner.

**Comment by valencebond** [comment]
thanks~

**Comment by valencebond** [comment]
hi @voyazici i read your paper again, but there is no content about table2 experimental settings, whether using SWA or not.  And there is also no ablation study about SWA method. i think is important because SWA is not your contribution. Can you provide your experimental settings of best performance without SWA and the encoder+decoder baseline performance?

**Comment by voyazici** [comment]
Hi,
As I said in the previous comment, we did not use the SWA for the experiments whose results can be seen in Table 2. These experiments were conducted to show that BCE models yield actually better results than the fixed order approaches when fully trained. We include the SWA when we compare our model with the other papers.
Here are the results that I got for the PLA without SWA:
```
MACRO prec 78.85%, recall 66.71%, f1 72.28%
MICRO prec 81.31%, recall 70.90%, f1 75.75%,
```
and for the freq-first:
```
MACRO prec 79.04%, recall 61.68%, f1 69.29%
MICRO prec 81.18%, recall 68.05%, f1 74.04%
```
To reproduce these results you should launch the following commands:
```
python train_lstm.py -decoder_lr 1e-3 -encoder_lr 1e-5 -epochs 25 -save_path <save_path> -order_free pla -image_path
<image_path> -encoder_weights <encoder_weights> -dropout 0.25
```
```
python train_lstm.py -decoder_lr 1e-3 -encoder_lr 1e-5 -epochs 30 -save_path <save_path> -image_path
<image_path> -encoder_weights <encoder_weights> -dropout 0.25 -sort_by_freq
```
